src/machine.rs

```diff
@@ -398,6 +401,15 @@ impl<'mir, 'tcx> Evaluator<'mir, 'tcx> {
             preemption_rate: config.preemption_rate,
             report_progress: config.report_progress,
             since_progress_report: 0,
+            external_c_so_file: match &config.external_c_so_file {
+                Some(lib_file_path) =>
+                    Some(unsafe {
```

Comment:
document that this is safe in the sense that it's the caller's obligation to provide a correct shared object